benches/benches.rs

```diff
@@ -36,25 +39,95 @@ fn criterion_benchmark(c: &mut Criterion) {
+    let factorial_runner = load_contract_for_vm("programs/benches/factorial_2M.cairo");
+    let fibonacci_runner = load_contract_for_vm("programs/benches/fib_2M.cairo");
+    let logistic_map_runner = load_contract_for_vm("programs/benches/logistic_map.cairo");
+
+    let factorial_function = factorial_runner
+        .find_function("main")
+        .expect("failed to find main factorial function");
+    let fibonacci_function = fibonacci_runner
+        .find_function("main")
+        .expect("failed to find main fibonacci function");
+    let logistic_map_function = logistic_map_runner
+        .find_function("main")
+        .expect("failed to find main logistic map function");
+
+    {
+        let mut factorial_group = c.benchmark_group("factorial_2M");
+
+        factorial_group.bench_function("Cached JIT", |b| {
+            b.iter(|| jit_factorial.invoke_dynamic(factorial_function_id, &[], Some(u128::MAX)));
+        });
+        factorial_group.bench_function("Cached AOT", |b| {
+            b.iter(|| aot_factorial.invoke_dynamic(factorial_function_id, &[], Some(u128::MAX)));
+        });
+
+        factorial_group.bench_function("VM", |b| {
+            b.iter(|| {
+                factorial_runner.run_function_with_starknet_context(
+                    factorial_function,
+                    &[],
+                    Some(usize::MAX),
+                    StarknetState::default(),
+                )
+            });
+        });
+
+        factorial_group.finish();
+    }
+
+    {
+        let mut fibonacci_group = c.benchmark_group("fibonacci_2M");
+
+        fibonacci_group.bench_function("Cached JIT", |b| {
+            b.iter(|| jit_fibonacci.invoke_dynamic(fibonacci_function_id, &[], Some(u128::MAX)));
+        });
+        fibonacci_group.bench_function("Cached AOT", |b| {
+            b.iter(|| aot_fibonacci.invoke_dynamic(fibonacci_function_id, &[], Some(u128::MAX)));
+        });
+        fibonacci_group.bench_function("VM", |b| {
+            b.iter(|| {
+                fibonacci_runner.run_function_with_starknet_context(
+                    fibonacci_function,
+                    &[],
+                    Some(usize::MAX),
+                    StarknetState::default(),
+                )
+            });
+        });
+
+        fibonacci_group.finish();
+    }
+
+    {
+        let mut logistic_map_group = c.benchmark_group("logistic_map");
+
+        logistic_map_group.bench_function("Cached JIT", |b| {
+            b.iter(|| {
+                jit_logistic_map.invoke_dynamic(logistic_map_function_id, &[], Some(u128::MAX))
+            });
+        });
+
+        logistic_map_group.bench_function("Cached AOT", |b| {
+            b.iter(|| {
+                aot_logistic_map.invoke_dynamic(logistic_map_function_id, &[], Some(u128::MAX))
+            });
+        });
+
+        logistic_map_group.bench_function("VM", |b| {
+            b.iter(|| {
+                logistic_map_runner.run_function_with_starknet_context(
```

Comment:
A doubt I got seeing the benchmark numbers, are we sure this program is not failing?
We are not checking the result of the run_function_with_starknet_context call

Reply:
I've added some checks to assert this. It seems to be working properly.